### DISCUSSION TITLE
Is there any way to clear input component field through other component(button) event?

### DISCUSSION THREAD

**Discussion by martiniifun** [reflex-dev/reflex#301] [Q&A]
Thank you for your hustling effort.
Is there any way to clear input component field 
through other component(button) event?
(Like, Todo list in your gallery examples)

I've tried several ways, but couldn't find a solution.

**Comment by FHU-yezi** [comment]
Sure, just bind the content of input component to a State Var, and set it to an empty string in the button's event handler.

Tell us if this work, happy Pyneconing.

**Reply by martiniifun** [reply]
Yes, I've tried that way.
When I bind State Var with input value,
Input component goes freezing..
Maybe I haven't understand your advice..
I wrote my stupid code as below,
I beg, could you please comment to this?🥲

```python3
import pynecone as pc

class State(pc.State):
    input_string: str
    def clear_input(self):
        self.input_string = ""

def index():
    return pc.vstack(
        pc.input(value=State.input_string),
        pc.button(on_click=State.clear_input),
    )

app = pc.App(state=State)
app.add_page(index)
app.compile()
```

**Reply by picklelo** [reply]
You need to bind the `on_change` event trigger to allow the input to change:
```python
class State(pc.State):
    text: str

    def clear_input(self):
        self.text = ""

def index():
  return pc.vstack(
      pc.input(value=State.text, on_change=State.set_text),
      pc.button("Clear Input", on_click=State.clear_input),
  )
  ```
